### ISSUE TITLE
Fix false positives regression in `named-grid-areas-no-invalid` 

### ISSUE THREAD

**Issue by ai** [stylelint/stylelint#6041]
### What steps are needed to reproduce the bug?

```
$ pnpm add stylelint@14.8.0
$ echo "body { grid-template: 1fr / auto 1fr auto }" > test.css
$ echo '{ "rules": { "named-grid-areas-no-invalid": true } }' > .stylelintrc
$ pnpm stylelint test.css

test.css
 1:23  ✖  Expected same number of cell tokens in    named-grid-areas-no-invalid
          each string
```

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "named-grid-areas-no-invalid": true
  }
}
```

### How did you run Stylelint?

stylelint test.css

### Which version of Stylelint are you using?

`14.8.0`

### What did you expect to happen?

No error

### What actually happened?

False positive error:

```
 1:23  ✖  Expected same number of cell tokens in    named-grid-areas-no-invalid
          each string
```

### Does the bug relate to non-standard syntax?

No

### Proposal to fix the bug

I saw this error just after update to `14.8` so it could be a regression

**Comment by ybiquitous** [comment]
@ai Thank you for reporting the issue and using the template. I can confirm the regression also on the [demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIBG0CeACY6DmAnAllALQAuMAtgA4A2AhmfOgIwBmu6A9OrQK4kTM23PgIC+AHQB2IADQgAzqkklaADwTh582eAiSW+bBuBT06cSFw9qMeRcYnJZsxcm1yMYnkJFauGLTyRJIQRPiSAG601IT26CRWMKboEpKiIKJAA).

I've labeled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by ybiquitous** [comment]
The cause seems the change in PR #5983: 👀 

https://github.com/stylelint/stylelint/pull/5983/files#diff-fb2387c00b7cdea515a799b21420dbf6725898783f743ae70f40e74fb4cdb6d8

**Comment by ybiquitous** [comment]
@jeddy3 Thanks for creating the quick fix #6042. I believe the fix is enough to suppress the regression. 👍🏼 

---

I've looked into the real cause of the regression. See the change below of the `isRectangular()` utility.

*Previous*:

https://github.com/stylelint/stylelint/blob/3fc7fa0b00d77d78f5edc5769e67e699312b9330/lib/rules/named-grid-areas-no-invalid/utils/isRectangular.js#L8-L10

*Current*:

https://github.com/stylelint/stylelint/blob/74b35b70d6d4b947d131d2298a4da37e7e4cdc29/lib/rules/named-grid-areas-no-invalid/utils/isRectangular.js#L8-L14

When the `area` argument is an empty array,
- the previous version returned `true`
- the current version returns `false` instead

I was careless with the behavior of the `Array.prototype.every()` method. [MDN says](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every#description) about the `Array` method:

> `every` acts like the "for all" quantifier in mathematics. In particular, for an empty array, it returns `true`.

In summary, I believe the current behavior of `isRectangular()` and the #6042 change (that checks `array.length` before calling `isRectangular`) is reasonable.
